test/integration/events/events_test.go

```diff
@@ -73,28 +73,32 @@ func TestEventCompatibility(t *testing.T) {
 	newRecorder := newBroadcaster.NewRecorder(scheme.Scheme, "k8s.io/kube-scheduler")
 	newBroadcaster.StartRecordingToSink(stopCh)
 	newRecorder.Eventf(regarding, related, v1.EventTypeNormal, "memoryPressure", "killed", "memory pressure")
-	err = wait.PollImmediate(100*time.Millisecond, 20*time.Second, func() (done bool, err error) {
```

Comment:
We should move this to https://pkg.go.dev/k8s.io/apimachinery/pkg/util/wait#PollUntilContextTimeout as opposed to rewriting it to not use the apimachinery

Reply:
actually the issue was to replace the polling based mechanism to a  interrupt based mechanism